### PR TITLE
Format triage data as arrays of dicts, instead of arrays of arrays.

### PR DESCRIPTION
**Pull request by rmmh**
This is much easier to handle in statically typed languages, and avoids
having to remember arbitrary array indices to write code effectively.

Before:
```python
  [['some awful stack trace exit 1', random_hash_1, 'some awful stack trace exit 1',
    [['example test', [['some-job', [1, 2, 3, 4]]]]]
   ],
   ['some other error message', random_hash_2, 'some other error message',
    [['unrelated test', [['other-job', [5, 7]]]],
     ['example test', [['some-job', [4]]]]]
   ]
  ]
```

After:
```python
  [{'id': random_hash_1,
    'key': 'some awful stack trace exit 1',
    'tests': [{'jobs': [{'builds': [1, 2, 3, 4],
                         'name': 'some-job'}],
               'name': 'example test'}],
    'text': 'some awful stack trace exit 1'},
   {'id': random_hash_2,
    'key': 'some other error message',
    'tests': [{'jobs': [{'builds': [5, 7],
                         'name': 'other-job'}],
               'name': 'unrelated test'},
              {'jobs': [{'builds': [4], 'name': 'some-job'}],
               'name': 'example test'}],
    'text': 'some other error message'}]
```

This increases the JSON size by ~2% before compression, and <1% after.

### CODE REVIEW

**Comment by fejta** [comment]
Can you put a before/after json object example in this PR description?

File: triage/Makefile [hunk 1]
```diff
@@ -25,6 +25,9 @@ push:
 	gcloud docker -- push $(IMG):latest
 	echo Pushed $(IMG):latest and $(IMG):$(TAG)
 
-.PHONY: all build push
+push_static:
```
Comment (fejta) [inline triage/Makefile]: Create a postsubmit prow job to do this on merge?

File: triage/summarize.py [hunk 2]
```diff
@@ -272,13 +272,18 @@ def tests_group_by_job(tests, builds):
 
 def clusters_to_display(clustered, builds):
     """Transpose and sort the output of cluster_global."""
-    return [
-        [key, key_id, clusters[0][1][0]['failure_text'],
-            [
-                [test_name, tests_group_by_job(tests, builds)]
+    return [{
```
Comment (fejta) [inline triage/summarize.py]: Looks like this is the primary change?

**Comment by fejta** [comment]
Will let @cjwagner apply the `/lgtm`

**Comment by spxtr** [comment]
Looks like this broke us: https://k8s-testgrid.appspot.com/test-infra

**Pull request merged**